pkg/dashboards/folders.go

```diff
@@ -0,0 +1,883 @@
+		log.Errorf("deleteFolder: Failed to upload query nodes dir, err=%v", err)
+		return err
+	}
+
+	return nil
+}
+
+func collectItemsToDelete(folderID string, structure *FolderStructure, itemsToDelete *[]string) {
+	// Add current folder to delete list
+	*itemsToDelete = append(*itemsToDelete, folderID)
+
+	// Recursively process children
+	if childIDs, exists := structure.Order[folderID]; exists {
+		for _, childID := range childIDs {
+			if child, exists := structure.Items[childID]; exists {
+				if child.Type == "folder" {
+					collectItemsToDelete(childID, structure, itemsToDelete)
+				} else {
+					*itemsToDelete = append(*itemsToDelete, childID)
+				}
+			}
+		}
+	}
+}
+
+func deleteDashboardFile(dashboardID string) error {
+	dashboardDetailsFname := config.GetDataPath() + "querynodes/" + config.GetHostID() + "/dashboards/details/" + dashboardID + ".json"
+	return os.Remove(dashboardDetailsFname)
+}
+
+func listItems(req *ListItemsRequest) (*ListItemsResponse, error) {
+	// Read folder structure
+	structure, err := readCombinedFolderStructure()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read folder structure: %v", err)
+	}
+
+	items := make([]ItemMetadata, 0)
+
+	getFullPath := func(itemID string) string {
+		path := []string{}
+		currentID := itemID
+		for currentID != "" && currentID != rootFolderID {
+			if item, exists := structure.Items[currentID]; exists {
+				path = append([]string{item.Name}, path...)
+				currentID = item.ParentID
+			} else {
+				break
+			}
+		}
+		return strings.Join(path, "/")
+	}
+
+	for id, item := range structure.Items {
+		// Skip root folder
+		if id == rootFolderID {
+			continue
+		}
+
+		// Apply folder filter
+		if req.FolderID != "" {
+			if req.FolderID == rootFolderID {
+				// For root folder, include all items at any level
+				// No filtering needed as we want to show everything
+			} else {
+				// For specific folder, show only immediate children
+				if item.ParentID != req.FolderID {
+					continue
+				}
+			}
+		}
+
+		// Apply type filter
+		if req.Type != "" && req.Type != "all" && item.Type != req.Type {
+			continue
+		}
+
+		// Get item details
+		details, _ := getDashboard(id)
+		isStarred := false
+		var createdAt time.Time
+		description := ""
+		if item.Type == "dashboard" {
+			if details != nil {
+				if starred, ok := details["isFavorite"].(bool); ok {
+					isStarred = starred
+				}
+				if created, ok := details["createdAt"].(float64); ok && created > 0 {
+					createdAt = time.UnixMilli(int64(created))
+				} else if created, ok := details["createdAt"].(int64); ok && created > 0 {
+					createdAt = time.UnixMilli(created)
+				}
+				if desc, ok := details["description"].(string); ok {
+					description = desc
+				}
+			}
+		} else if item.Type == "folder" {
+			if item.CreatedAt > 0 {
+				createdAt = time.UnixMilli(item.CreatedAt)
+			}
+		}
+
+		// Apply starred filter
+		if req.Starred && !isStarred {
+			continue
+		}
```

Comment:
You might already be handling this, but I have a question. It seems like there's three things users might want to search for: only starred, only non-starred, and all. Is there a way for them to make searches for each of these cases?